lib/blockscore/base.rb

```diff
@@ -0,0 +1,59 @@
+require 'active_support/core_ext/class'
+require 'active_support/core_ext/string/inflections'
+require 'blockscore/connection'
+
+module BlockScore
+  class Base
+    extend BlockScore::Connection
+
+    class_attribute :api_key, :resource, :version
+
+    def self.inherited(base)
+      base.resource = base.to_s.split('::').last.underscore
+    end
+
+    def initialize(options = {})
+      options[:class] = resource
+      @attrs = options
+    end
+
+    def refresh
+      self.class.retrieve(id)
+      instance_variable_set(:@attrs, r.instance_variable_get(:@attrs))
```

Comment:
@backus Actually, do we want to make the `attr_reader` for `@attrs` and make it publicly available like that? It seems like a bit of a leaky abstraction as users of the gem shouldnt be able to access `@attrs` directly, right?